### PR TITLE
datatrails: feat: metadata for metric panels

### LINKED ISSUE #79596: Data Trails: Add metrics metadata
(Advanced options can be separate from this issue)

This issue focuses on adding an info icon with the metrics description. Note: If a metric has no description, we can hide the info icon

Figma file: https://www.figma.com/file/k87knv2idoa0EIsLpQOWkR/Data-Trails?type=design&node-id=418%3A13135&mode=design&t=QNUOVenWOtsVDmMI-1

Image

### PR DESCRIPTION
**Pull request by darrenjaneczek**
**What is this feature?**

Use the metric metadata to create a description for the panels' info tooltip.
image_720
image_720
image_720

Centralizes metadata retrieval to a helper object attached to the DataTrail object.

- Fixes: https://github.com/grafana/grafana/issues/79596

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trails/MetricSelectScene.tsx [hunk 1]
```diff
@@ -241,6 +244,15 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
...
         }
         const panel = getPreviewPanelFor(metric.name, index, currentFilterCount);
+
+        const vizPanel = panel.state.body;
+        if (vizPanel instanceof VizPanel) {
+          trail.getMetricMetadata(metric.name).then((metadata) => {
+            const description = getMetricDescription(metadata);
+            vizPanel.setState({ description });
+          });
+        }
```
Comment (torkelo) [inline public/app/features/trails/MetricSelectScene.tsx:254]: I would make this function async and wait for await call, and pass in the description for getPreviewPanelFor , that way we don't have this double rendering. the metadata call is fast (and should be cached)

**Comment by darrenjaneczek** [comment]
Also added the info description to the non-preview mode panels:

image

File: public/app/features/trails/MetricSelectScene.tsx (hunk 1 above)
Comment (darrenjaneczek) [reply public/app/features/trails/MetricSelectScene.tsx]: Fixed

**Comment by catherineymgui** [comment]
- Is it possible to hide the info icon on the main metric graph? The metadata is mostly useful when users are comparing metrics and would want to learn more to determine if it's the one they want to select.
- Would also flip around the description and type. Have the description have normal styling.
Screenshot 2024-03-21 at 11 41 03

**Comment by darrenjaneczek** [comment]
Info icon no longer on main selected metric:
image

Changed format of description:
image

Note that if a unit was reported, it would appear below the type, in the same format:
```ts
  const lines = [
    help, //
    type && `**Type:** *${type}*`,
    unit && `**Unit:** ${unit}`,
  ];
```

**Comment by catherineymgui** [comment]
Looks good to me!

**Review by adrapereira — APPROVED** [review]

**Pull request merged**